File: src/main/dev/danzou/advent22/kotlin/Day1.kt
package dev.danzou.advent22.kotlin

import dev.danzou.advent22.AdventTestRunner22

internal class Day1 : AdventTestRunner22() {
    override fun part1(input: String): Number =
        input.split("\n\n")
            .map { it.split("\n").map { it.toLong() } }
            .map { it.sum() }
            .max()

    override fun part2(input: String): Number =
        input.split("\n\n")
            .map { it.split("\n").map { it.toLong() } }
            .map { it.sum() }
            .sorted()
            .takeLast(3)
            .sum()
}

File: src/main/dev/danzou/advent22/scala/Day1.scala
package dev.danzou.advent22.scala

import dev.danzou.advent22.AdventTestRunner22

object Day1 extends AdventTestRunner22 {
  override def part1(input: String): AnyRef = ???

  override def part2(input: String): Number = input
    .split("\n\n")
    .map(_.split("\n").map(_.toLong))
    .map(_.sum)
    .sorted
    .takeRight(3)
    .sum
}